rdmd.d

```diff
@@ -693,10 +693,16 @@ private string[string] getDependencies(string rootModule, string workDir,
 
     immutable rootDir = dirName(rootModule);
 
-    // In most cases we don't have any dependencies except the standard library
+    // For most single-file builds we don't have any dependencies except the standard library
     auto file = readText(rootModule);
-    if (!needsInspection(file))
-        return null;
+    bool hasDependencies = needsInspection(file);
+    if (!hasDependencies)
+    {
+        import std.file : writeFile = write;
+        // run dmd with an empty file to get it's configuration
+        rootModule = buildPath(workDir, "emptyFile.d");
+        rootModule.writeFile("void main(){}");
+    }
```

Comment:
Well then I have to modify the lines below. As we have to write an empty file anyways (DMD expects at least one file) , I thought this was the option with the least changes.